webbpsf/tests/test_roman.py

```diff
@@ -185,23 +185,12 @@ def test_swapping_modes(wfi=None):
     if wfi is None:
         wfi = roman.WFI()
 
-    # change detector string to match file format (e.g., "SCA01" -> "SCA_1")
-    detector_substr = lambda det: f'{det[:3]}_{str(int((det[3:])))}'
-
-    # dynamically generate current pupil path for a given WFI instance
-    pupil_path = lambda self, mask=None: os.path.join(
-        self._pupil_controller._pupil_basepath,
-        self._pupil_controller.pupil_file_formatters[
-            self._pupil_controller._get_filter_mask(self.filter) if mask is None else mask
-        ],
-    ).format(detector_substr(self.detector))
-
     tests = [
         # [filter, mode, pupil_file]
-        ['F146', 'imaging', pupil_path],
-        ['F213', 'imaging', pupil_path],
-        [PRISM_FILTERS[0], 'prism', pupil_path],
-        [GRISM_FILTERS[0], 'grism', pupil_path],
+        ['F146', 'imaging', pupil_path(wfi)],
```

Comment:
I don't follow this edit. I see the lambda for pupil_path was deleted from the existing version, but I don't see a replacement implementation in the revised file?

Reply:
the lambda was a redefine of `def pupil_path` on line 22